### PR TITLE
feat/custom analysis

### PR DESCRIPTION
**Pull request by AlexsJones**
Closes # 

## ✅ Checks

- [x ] My pull request adheres to the code style of this project
- [x ] My code requires changes to the documentation
- [x ] I have updated the documentation as required
- [ x] All the tests have passed

### CODE REVIEW

File: pkg/client/config.go [hunk 1]
```diff
@@ -54,9 +67,10 @@ func (c *Client) AddConfig(config *v1alpha1.K8sGPT) error {
 
 func (c *Client) RemoveConfig(config *v1alpha1.K8sGPT) error {
```
Comment (JuHyung-Son) [inline pkg/client/config.go:68]: Does this function actually work?
RemoveConfigRequest is actually an empty value.

**Review by JuHyung-Son — APPROVED** [review]

File: pkg/client/config.go (hunk 1 above)
Comment (arbreezy) [reply pkg/client/config.go:68]: RemoveConfig is actually only related to removing the cache configuration, frankly speaking the whole config API that we have created needs better structuring as it has grown organically including different functionality

It _works_  by removing whatever is configured in the k8sgpt's configuration cache stanza..

So neither integrations nor now the customanalyzers will be deleted with removeConfig()

**Comment by arbreezy** [comment]
We also have to add the new crd schema to the helm templates

**Review by arbreezy — DISMISSED** [review]

File: pkg/client/config.go (hunk 1 above)
Comment (AlexsJones) [reply pkg/client/config.go:68]: Yeah we were just discussing doing a refactor of the config stuff in slack

**Pull request merged**